tools/semgrep/ci/loop-time-after.yml

```
@@ -0,0 +1,14 @@
+rules:
+  - id: loop-time-after
+    pattern: |
+      for {
+        select {
```

I wonder if we also want to look for more general occurrences of this pattern. This seems specifically focused on the case where `select` immediately follows in the for loop. I think I see some less-obvious occurrences of this pattern:

```suggestion
        ...
        select {
```

Modifying the rule with that suggestion produces 4 more results that we may/may not want to consider. To take this point further, can this be generalized to include other patterns, like `time.After()` embedded in anonymous functions?

---

switched to a more general version of the rule that matches additional results